### PR TITLE
[WIP] Extract Select component

### PR DESCRIPTION
**Pull request by NehaBroad**
### What
- 

### Why
- 

### Testing strategy
- [ ]

### CODE REVIEW

File: packages/components/src/Select.ts
Comment (nawatts) [inline packages/components/src/Select.ts:1]: We've recently (#4637) intentionally removed react-hyperscript-helpers from this package. This component would need to be updated to use JSX.

File: packages/components/src/Select.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  Props as RSelectProps,
+  PropsValue as RSelectPropsValue,
+  SingleValue as RSelectSingleValue,
+} from 'react-select';
+
+import { useUniqueId } from './hooks/useUniqueId';
+import { icon } from './icon';
+import { useThemeFromContext } from './theme';
+
+let colors: any;
```
Comment (nawatts) [inline packages/components/src/Select.ts:16]: Can we find a better way than a global variable to pass the colors from the theme context through to React Select's theme?

File: packages/components/src/hooks/useOnMount.ts
Comment (nawatts) [inline packages/components/src/hooks/useOnMount.ts:1]: My experience with this function is that it almost always results in a bug. Thus, I'm strongly opposed to promoting it to the components package. It doesn't look like it's required for this PR, but if so, it should be replaced with `useEffect`, where we get the benefit of ESLint checking the dependencies list.

**Review by nawatts — CHANGES_REQUESTED** [review]

**Comment by NehaBroad** [comment]
Thank you @nawatts ! I am going to close this PR, this was just a test to see if i can open a PR as i did not have permissions. Its still WIP i'll re-open this with final changes

**Pull request closed without merging**